core/market/migrations/2020-10-04-190200_agreement_events/up.sql

```diff
@@ -0,0 +1,87 @@
+-- Your SQL goes here
+
+CREATE TABLE market_agreement_event(
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    agreement_id INTEGER NOT NULL,
+    event_type VARCHAR(10) NOT NULL,
+    timestamp DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+    issuer VARCHAR(1) NOT NULL,
+    reason TEXT,
+    signature TEXT,
+
+    FOREIGN KEY(agreement_id) REFERENCES market_agreement (id),
+    UNIQUE(agreement_id, event_type)
+    CHECK (event_type in ('Terminated', 'Approved', 'Cancelled', 'Rejected'))
+    CHECK (issuer in ('P', 'R'))
+);
+
+-- Add session_id. Will drop all Agreements.
+DROP TABLE market_agreement;
+DROP TABLE agreement_state;
+
+CREATE TABLE market_agreement(
+    id VARCHAR(100) NOT NULL PRIMARY KEY,
+
+    demand_properties TEXT NOT NULL,
+    demand_constraints TEXT NOT NULL,
+
+    offer_properties TEXT NOT NULL,
+    offer_constraints TEXT NOT NULL,
+
+    offer_id VARCHAR(97) NOT NULL,
+    demand_id VARCHAR(97) NOT NULL,
+
+    offer_proposal_id VARCHAR(100) NOT NULL,
+    demand_proposal_id VARCHAR(100) NOT NULL,
+
+    provider_id VARCHAR(20) NOT NULL,
+    requestor_id VARCHAR(20) NOT NULL,
+
+    session_id VARCHAR(100),
+
+    creation_ts DATETIME NOT NULL,
+    valid_to DATETIME NOT NULL,
+    state VARCHAR(20) NOT NULL,
+    approved_date DATETIME,
```

Comment:
btw.  maybe we can make the name consistent
```suggestion
    approved_ts DATETIME,
```